lib/src/main/java/com/diffplug/spotless/FileSignature.java

```diff
@@ -77,21 +78,21 @@ public static FileSignature signAsSet(File... files) throws IOException {
 
 	/** Creates file signature insensitive to the order of the files. */
 	public static FileSignature signAsSet(Iterable<File> files) throws IOException {
-		return new FileSignature(toSortedSet(files));
+		return new FileSignature(toSortedSet(files, comparing(File::getName)));
```

Comment:
Right, yeah this is a real problem. Is it maybe an option to sort on filename + filehash and keep a sorted list instead of a set? I do not know if removing duplicates is important or not.

Reply:
The only problem I can see with sorting on filename (and this is a problem we already have), would be if a collection of files had members from different root directories.  For example, a set of files like `'/tmp/otherjar', '/Users/a/.cache/jar'` vs `'/home/z/.cache/jar', '/tmp/otherjar'`.  But I *think* it's the case that files will mostly share the same root folders.  For example, `JarState` will always be `~/.m2` or `~/.gradle/cache`, and config files will always be in the project directory.  That's not exact though, and sorting only on filehash would be more "exact".

The `.gitattributes` FileSignature is probably going to need to be tweaked regardless.  All we care about are the line-endings, but we're sucking in all of the user's homedir `.gitconfig` customizations right now.

Reply:
I think removing duplicates does matter for `signAsSet`, but the old logic did a good job using a list and just removing the duplicates.

Reply:
I think the `.gitattributes` problem is solved by #621, which revives this PR as a great solution.  If we can fix the other nits, then I think we can merge and release this.

Reply:
Thanks for the update. I'll rebase this PR and address all the comments this week.